apiserver/facades/client/sshclient/facade.go

```diff
@@ -358,18 +358,19 @@ func (facade *Facade) PublicHostKeyForTarget(arg params.SSHHostKeyRequestArg) pa
 		return res
 	}
 
-	var hostkey string
+	var knownHostLine string
+
 	switch info.Target() {
 	case virtualhostname.MachineTarget:
 		machineId, _ := info.Machine()
-		hostkey, err = facade.backend.MachineVirtualPublicHostKeyPEM(strconv.Itoa(machineId))
+		knownHostLine, err = facade.backend.MachineVirtualHostKeyKnownHostLine(strconv.Itoa(machineId), arg.Hostname)
```

Comment:
I expect there will be 2 use cases for the this facade method.
1. User runs `juju ssh unit/0` and the CLI needs to add the server's public key to it's known-hosts file.
2. User run `juju ssh unit/0 --print-command` (or whatever the flag is) and that needs to print the hash of the server's public key that the user can verify when they use their SSH client, and ones like openssh will then add the details to the known-hosts file.

All that being said. We should probably just return both the knownHosts line, and the fingerprint of the key to cater for both flows.